x-pack/legacy/plugins/reporting/index.ts

```diff
@@ -90,18 +81,22 @@ export const reporting = (kibana: any) => {
         config: server.config,
         info: server.info,
         route: server.route.bind(server),
-        plugins: {
-          elasticsearch: server.plugins.elasticsearch,
-          xpack_main: server.plugins.xpack_main,
-        },
+        plugins: { xpack_main: server.plugins.xpack_main },
         savedObjects: server.savedObjects,
         fieldFormatServiceFactory,
         uiSettingsServiceFactory: server.uiSettingsServiceFactory,
       };
 
-      const initializerContext = server.newPlatform.coreContext;
-      const plugin: ReportingPlugin = reportingPluginFactory(initializerContext, __LEGACY, this);
-      await plugin.setup(coreSetup, pluginsSetup);
+      const plugin: ReportingPlugin = reportingPluginFactory(
+        server.newPlatform.coreContext,
+        __LEGACY,
+        this
+      );
+      await plugin.setup(coreSetup, {
+        elasticsearch: coreSetup.elasticsearch,
+        security: server.newPlatform.setup.plugins.security as SecurityPluginSetup,
+        usageCollection: server.newPlatform.setup.plugins.usageCollection,
+      });
```

Comment:
Redundant parameters: `elasticsearch` is already provided by `coreSetup`, `elasticsearch: coreSetup.elasticsearch,` seems unnecessary.

Reply:
The params are forced to fit the `ReportingStartDeps` interface, which is used all over the place to pull the ES service. Unfortunately it seems to make better sense to keep it this way, but I could see it changing later on.